File: src/main/java/com/home/demos/samplebankmodule/model/Card.java
package com.home.demos.samplebankmodule.model;

import com.home.demos.samplebankmodule.config.SampleBankModuleConfiguration;
import lombok.Data;

import java.util.List;

@Data
public class Card {
    private Long id;
    private String title;
    private Long clientId;
    private Long sum;

    public Card(Long clientId) {
        this.clientId = clientId;
    }

    public Card() {
    }

    public Card create() {
        return SampleBankModuleConfiguration
                .cardRepository()
                .save(this);
    }

    public List<Card> findAllLikeThis() {
        return SampleBankModuleConfiguration
                .cardRepository()
                .findAllByClientId(this.clientId);
    }
}

File: src/main/kotlin/com/home/demos/samplebankmodule/model/Card.kt
package com.home.demos.samplebankmodule.model

import com.home.demos.samplebankmodule.config.SampleBankModuleConfiguration.Companion.asyncRepository
import com.home.demos.samplebankmodule.config.SampleBankModuleConfiguration.Companion.cardRepository

class Card(
        val id: Long,
        val title: String,
        var clientId: Long,
        val sum: Long
) {

    fun create(): TransactionId {
        return asyncRepository().process(this) { cardRepository().save(this) }
    }

    fun findAllLikeThis(): TransactionId {
        return asyncRepository().process(this) { cardRepository().findAllByClientId(clientId) }
    }
}
